ldap/src/main/java/org/openmbee/mms/ldap/LdapSecurityConfig.java

```diff
@@ -178,4 +189,28 @@ public BaseLdapPathContextSource contextSource() {
         return contextSource;
     }
 
+    private User saveLdapUser(DirContextOperations userData, User saveUser) {
+        if (!saveUser.getEmail().equals(userData.getStringAttribute(userAttributesEmail))) {
+            saveUser.setEmail(userData.getStringAttribute(userAttributesEmail));
+        }
+        if (!saveUser.getFirstName().equals(userData.getStringAttribute(userAttributesFirstName))) {
+            saveUser.setFirstName(userData.getStringAttribute(userAttributesFirstName));
+        }
+        if (!saveUser.getLastName().equals(userData.getStringAttribute(userAttributesLastName))) {
+            saveUser.setLastName(userData.getStringAttribute(userAttributesLastName));
+        }
+
+        return saveUser;
+    }
+
+    private User createLdapUser(DirContextOperations userData) {
```

Comment:
should also try to init first and last name on create here? otherwise it'll be filled in if they log in again after 24 hours